src/vs/platform/telemetry/node/workbenchCommonProperties.ts

```diff
@@ -51,3 +53,23 @@ export function resolveWorkbenchCommonProperties(storageService: IStorageService
 // 	storageService.store('telemetry.instanceId', result);
 // 	return result;
 // }
+
+// {{SQL CARBON EDIT}}s
+function setUsageDates(storageService: IStorageService): void {
+	// daily last usage date
+	const dailyLastUseDate = storageService.get('telemetry.dailyLastUseDate') || new Date().toUTCString();
```

Comment:
if we initialize the last use date to the current date won't we not send any events the first time someone uses the product?  Should we initialize to a date that is guaranteed to be at least a month in the past (you could just use a fixed date like `Jan 1st, 2000`, or something like that).